tests/phpunit/integration/Modules/Thank_With_GoogleTest.php

```diff
@@ -39,32 +46,123 @@ public function set_up() {
+			Settings::OPTION,
+			array(
+				'publicationID'   => '12345',
+				'colorTheme'      => 'blue',
+				'buttonPlacement' => 'static_auto',
+				'buttonPostTypes' => array( 'post' ),
+			)
+		);
+	}
+
+	public function test_register_template_redirect() {
+		remove_all_actions( 'template_redirect' );
+
+		$this->thank_with_google->register();
+
+		remove_all_actions( 'wp_enqueue_scripts' );
+		remove_all_filters( 'the_content' );
+
+		$this->assertFalse( has_action( 'wp_enqueue_scripts' ) );
+		$this->assertFalse( has_filter( 'the_content' ) );
+
+		do_action( 'template_redirect' );
+		$this->assertTrue( has_action( 'wp_enqueue_scripts' ) );
+		$this->assertTrue( has_filter( 'the_content' ) );
+
+		remove_all_actions( 'wp_enqueue_scripts' );
+		remove_all_filters( 'the_content' );
+
+		$this->assertFalse( has_action( 'wp_enqueue_scripts' ) );
+		$this->assertFalse( has_filter( 'the_content' ) );
+
+		// Tag not hooked when blocked.
+		add_filter( 'googlesitekit_' . Thank_With_Google::MODULE_SLUG . '_tag_blocked', '__return_true' );
+		do_action( 'template_redirect' );
+
+		$this->assertFalse( has_action( 'wp_enqueue_scripts' ) );
+		$this->assertFalse( has_filter( 'the_content' ) );
+	}
+
+	public function test_footer_snippet() {
+		remove_all_actions( 'template_redirect' );
+		remove_all_actions( 'wp_enqueue_scripts' );
+
+		$this->thank_with_google->register();
+
+		// Hook `wp_print_footer_scripts` on placeholder action for capturing.
+		add_action( '__test_print_scripts', 'wp_print_footer_scripts' );
+
+		do_action( 'template_redirect' );
+		do_action( 'wp_enqueue_scripts' );
+
+		$output = $this->capture_action( '__test_print_scripts' );
+
+		// Snippet should not be present on non singular pages.
+		$this->assertEmpty( $output );
+
+		$post_ID = $this->factory()->post->create();
+		$this->go_to( get_permalink( $post_ID ) );
+
+		do_action( 'template_redirect' );
+		do_action( 'wp_enqueue_scripts' );
+
+		$output = $this->capture_action( '__test_print_scripts' );
+
+		$this->assertStringContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringContainsString( 'buttonPosition: \'inline\',', $output );
+		$this->assertStringContainsString( 'isPartOfProductId: \'12345:default\',', $output );
+	}
+
+	public function test_content_placeholder() {
+		remove_all_filters( 'the_content' );
+
+		$this->thank_with_google->register();
+		do_action( 'template_redirect' );
+
+		$mock_content = '<p>Hello World</p>';
+
+		$output = apply_filters( 'the_content', $mock_content );
+
+		// Snippet should not be present on non singular pages.
+		$this->assertEquals( $output, $mock_content );
+
+		$post_ID = $this->factory()->post->create();
+		$this->go_to( get_permalink( $post_ID ) );
+
+		$output = apply_filters( 'the_content', '<p>Hello World</p>' );
+
+		$this->assertStringContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringContainsString( '<button twg-button', $output );
```

Comment:
This should be moved into a new test file for the Web_Tag class and more tests should be added to cover all three use cases.

Reply:
I have moved it to separate file and added more tests. However, the tests are failing on PHP 5.6 and/or WP 4.7. Can you help me with it?

Reply:
@kuasha420, not sure why it doesn't work, maybe try to add `'post_status'  => 'publish',` when you generate a new post?